scraper/scraper_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scraper
+
+import (
+	"crypto/tls"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+)
+
+func TestHandler(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		t.Parallel()
+
+		s := New(1*time.Second, log.NewNopLogger(), &tls.Config{})
```

Comment:
```suggestion
		s := New(1*time.Second, log.NewNopLogger(), nil)
```

We want no TLS in test, not default TLS

Reply:
(I tested this locally)

Reply:
Yea, I was just figuring that out as well. :)